desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py

```diff
@@ -171,6 +172,18 @@ def _create_engine(self):
       s3_staging_dir = url.rsplit('s3_staging_dir=', 1)[1]
       url = url.replace(s3_staging_dir, urllib_quote_plus(s3_staging_dir))
 
+    m = re.search(URL_PATTERN, url)
+    driver_name = m.group('driver_name')
+    if self.options.get('has_impersonation'):
+      if not driver_name:
+        raise QueryError('Driver name of %(url)s could not be found and impersonation is turned on' % {'url': url})
+
+      if driver_name.startswith("phoenix"):
```

Comment:
Nit: Maybe add a comment here or above line 199 on why we are explicitly checking for phoenix? Like `phoenixdb does not support impersonation using principal_username parameter`